core/eolearn/tests/test_core_tasks.py

```diff
@@ -81,6 +82,83 @@ def test_add_rename_remove_feature(self):
+
+        self.assertTrue(np.array_equal(patch.mask[duplicate_name], mask_data),
+                        'Feature was not duplicated correctly. Data does not match.')
+
+        with self.assertRaises(ValueError, msg='Expected a ValueError when creating an already exising feature.'):
+            duplicate_task(patch)
+
+        duplicate_names = {'D1', 'D2'}
+        feature_list = [(FeatureType.MASK, 'MASK1', 'D1'), (FeatureType.MASK, 'MASK2', 'D2')]
+        patch = DuplicateFeature(feature_list).execute(patch)
+
+        self.assertTrue(duplicate_names.issubset(patch.mask.keys()), 'Duplicating multiple features failed.')
+
+        # Duplicating MASK1 three times into D3, D4, D5 doesn't work, because EOTask.feature_gen
+        # returns a dict containing only ('MASK1', 'D5') duplication
+
+        # duplicate_names = {'D3', 'D4', 'D5'}
+        # feature_list = [(FeatureType.MASK, 'MASK1', new) for new in duplicate_names]
+        # patch = DuplicateFeature(feature_list).execute(patch)
+
+        # self.assertTrue(duplicate_names.issubset(patch.mask.keys()),
+        #                 'Duplicating single feature multiple times failed.')
+
+    def test_initialize_feature(self):
+        patch = DeepCopyTask()(self.patch)
+
+        init_val = 123
+        shape = (5, 10, 10, 3)
+        compare_data = np.ones(shape) * init_val
+
+        InitializeFeature((FeatureType.MASK, 'test'), shape=shape, init_value=init_val)(patch)
+        self.assertEqual(patch.mask['test'].shape, shape)
+        self.assertTrue(np.array_equal(patch.mask['test'], compare_data))
+
+        failmsg = 'Expected a ValueError when trying to initialize a feature with a wrong shape dmensions.'
+        with self.assertRaises(ValueError, msg=failmsg):
+            InitializeFeature((FeatureType.MASK_TIMELESS, 'wrong'), shape=shape, init_value=init_val)(patch)
+
+        init_val = 123
+        shape = (10, 10, 3)
+        compare_data = np.ones(shape) * init_val
+
+        InitializeFeature((FeatureType.MASK_TIMELESS, 'test'), shape=shape, init_value=init_val)(patch)
+        self.assertEqual(patch.mask_timeless['test'].shape, shape)
+        self.assertTrue(np.array_equal(patch.mask_timeless['test'], compare_data))
+
+        fail_msg = 'Expected a ValueError when trying to initialize a feature with a wrong shape dmensions.'
+        with self.assertRaises(ValueError, msg=fail_msg):
+            InitializeFeature((FeatureType.MASK, 'wrong'), shape=shape, init_value=init_val)(patch)
+
+        init_val = 123
+        shape = (5, 10, 10, 3)
+        compare_data = np.ones(shape) * init_val
+        new_names = {'F1', 'F2', 'F3'}
+
+        InitializeFeature({FeatureType.MASK: new_names}, shape=shape, init_value=init_val)(patch)
+        fail_msg = "Failed to initialize new features from a shape tuple."
+        self.assertTrue(new_names < set(patch.mask.keys()), msg=fail_msg)
```

Comment:
You don't need to write `.keys()` because in Python 3 you can iterate over keys of a dictionary simply by iterating over the dictionary itself.

Reply:
Done.

Reply:
Note for next PR: `.keys() appears a few more times in `test_core_tasks.py`.